### DISCUSSION TITLE
Logs getting printed in System.debug from Logger methods in Apex

### DISCUSSION THREAD

**Discussion by Kritarth-Sachdeva** [jongpie/NebulaLogger#583] [Q&A]
We are using v4.11.2 unmanaged package.

On writing below line in apex, logs are getting printed in System.debug as well, even though 'Enable Apex System.debug()' is unchecked in Logger settings.

Logger.info('test print log');

Is it possible to stop the printing of logs in System logs or will they always be printed, is any configuration change required for this?
Any guidance will be much appreciated.

**Comment by jongpie** [comment]
Hi @Kritarth-Sachdeva - so sorry for the late response! Were you able to figure out this issue? I tried to reproducing it, but everything seems to be working for me. I also reviewed the code for `v4.11.2`, and [it seems to have the correct check on the field `LoggerSettings__c.IsApexSystemDebugLoggingEnabled__c`](https://github.com/jongpie/NebulaLogger/blob/v4.11.12/nebula-logger/core/main/logger-engine/classes/LogEntryEventBuilder.cls#L753C1-L756C10), so I'm not sure what could be the cause of the issue.

If you're still experiencing this issue, could you try running this anonymous Apex script in your org & share the results?

```apex
System.debug('settings.IsApexSystemDebugLoggingEnabled__c== ' + Logger.getUserSettings().IsApexSystemDebugLoggingEnabled__c);
System.debug('settings.LoggingLevel__c== ' + Logger.getUserSettings().LoggingLevel__c);
Logger.info('test print log');
System.debug('just ran Logger.info(\'test print log\');');
```
